google/cloud/dataproc_spark_connect/session.py

```diff
@@ -208,12 +208,12 @@ def create_session_pbar():
                         ncols=80,
                     )
                     for i in pbar:
-                        if stop_create_session_pbar:
+                        if stop_create_session_pbar_event.is_set():
                             break
                         # Last iteration
                         if i >= iterations - 1:
                             # Sleep until session created
-                            while not stop_create_session_pbar:
+                            while not stop_create_session_pbar_event.is_set():
                                 time.sleep(1)
                         else:
                             time.sleep(1)
```

Comment:
I think this code can be simplified a bit:

```py
                    pbar = tqdm.trange(
                        iterations-1,
                        bar_format="{bar}",
                        ncols=80,
                    )
                    for i in pbar:
                        if stop_create_session_pbar_event.is_set():
                            return
                        time.sleep(1)
                    # Last iteration
                    while not stop_create_session_pbar_event.is_set():
                        time.sleep(1)
```

Reply:
Idea here is to not "finish" pbar and hold it in-proggress while it's not finished - I think that simplified code will wait after pbar already finished.